core/trino-main/src/main/java/io/trino/operator/project/MergePages.java

```diff
@@ -204,10 +205,17 @@ private void appendPage(Page page)
         {
             pageBuilder.declarePositions(page.getPositionCount());
             for (int channel = 0; channel < types.size(); channel++) {
-                Type type = types.get(channel);
-                for (int position = 0; position < page.getPositionCount(); position++) {
-                    type.appendTo(page.getBlock(channel), position, pageBuilder.getBlockBuilder(channel));
-                }
+                appendBlock(
+                        types.get(channel),
+                        page.getBlock(channel).getLoadedBlock(),
+                        pageBuilder.getBlockBuilder(channel));
+            }
+        }
+
+        private void appendBlock(Type type, Block block, BlockBuilder blockBuilder)
```

Comment:
is there any reason this unnecessariy-looking method wouldn't be inlined to make code more readable?

if you think a separate method does matter for more than just readability, add expanatory code comment please